packages/front-generator/src/generators/react-typescript/app/shared.ts

```diff
@@ -0,0 +1,212 @@
+import * as Base from "yeoman-generator";
+import * as path from "path";
+import {ProjectInfo, ProjectModel} from '../../../common/model/cuba-model';
+import {
+  ERR_STUDIO_NOT_CONNECTED, exportProjectModel,
+  getOpenedCubaProjects, normalizeSecret,
+  StudioProjectInfo
+} from '../../../common/studio/studio-integration';
+import {ownVersion} from '../../../cli';
+import {SdkAllGenerator} from '../../sdk/sdk-generator';
+// noinspection ES6PreferShortImport
+import {CommonGenerationOptions} from '../../../common/cli-options';
+// noinspection ES6PreferShortImport
+import {readProjectModel} from '../../../common/base-generator';
+import {throwError} from '../../../common/utils';
+import {SUPPORTED_CLIENT_LOCALES} from '../common/i18n';
+
+/**
+ * @alpha
+ */
+export interface ReactTSAppTemplateModel {
+  title: string;
+  basePath: string;
+  project: ProjectInfo;
+  ownVersion: string;
+}
+
+/**
+ * @alpha
+ */
+export interface ReactTSAppAnswers {
+  projectInfo: StudioProjectInfo;
+}
+
+/**
+ * @alpha
+ */
+export async function promptForProjectModel(
+  gen: Pick<Base, 'log' | 'env' | 'prompt'> & {
+    options: CommonGenerationOptions,
+  },
+  modelFilePath?: string
+): Promise<{
+  cubaProjectModel?: ProjectModel,
+  force?: boolean,
+  answers?: ReactTSAppAnswers
+}> {
+
+  // read model from path provided in options
+  if (gen.options.model) {
+    gen.log('Skipping prompts since model provided');
+    const cubaProjectModel = readProjectModel(modelFilePath);
+    return {
+      cubaProjectModel,
+      force: true
+    };
+  }
+
+  // obtain model from Studio
+  const openedCubaProjects = await getOpenedCubaProjects();
+  if (!openedCubaProjects || openedCubaProjects.length < 1) {
+    throwError(gen, ERR_STUDIO_NOT_CONNECTED);
+  }
+
+  const answers = await gen.prompt([{
+    name: 'projectInfo',
+    type: 'list',
+    message: 'Please select CUBA project you want to use for generation',
+    choices: openedCubaProjects && openedCubaProjects.map(p => ({
+      name: `${p.name} [${p.path}]`,
+      value: p
+    }))
+  }]) as ReactTSAppAnswers;
+
+  return {
+    answers
+  };
+}
+
+/**
+ * @alpha
+ */
+export async function prepareProjectModel(
+  gen: Pick<Base, 'env'> & {
+    answers?: ReactTSAppAnswers,
+  },
+  cubaProjectModel?: ProjectModel
+): Promise<{
+  model: ReactTSAppTemplateModel,
+  cubaProjectModel?: ProjectModel,
+  modelFilePath?: string,
+}> {
+  if (cubaProjectModel != null) {
+    return {
+      model: createModel(cubaProjectModel.project)
+    };
+  } else if (gen.answers != null) {
+    const modelFilePath = path.join(process.cwd(), 'projectModel.json');
+
+    await exportProjectModel(gen.answers.projectInfo.locationHash, modelFilePath);
+    // TODO exportProjectModel is resolved before the file is created. Timeout is a temporary workaround.
+    await new Promise(resolve => setTimeout(resolve, 2000));
+
```

Comment:
Looks like we can remove timeout, issue fixed in Studio https://youtrack.cuba-platform.com/issue/STUDIO-8257